text/0000-prefetch-hook.md

```diff
@@ -0,0 +1,79 @@
+- Start Date: 2015-10-01
+- RFC PR: (leave this empty)
+- Ember Issue: (leave this empty)
+
+# Summary
+
+The waterfall way in which Ember resolves a transition's model hooks can result in unnecessary blocking.
```

Comment:
some quick thoughts as i read.

Embers typical inter-route data-fetching is sequential, although this has upsides (simpler programming model) the resulting latency of each request is compounded and can result in a degraded experience. The general solution to the "waterfall" problem of sequential requests, is to execute independent requests in parallel.

Automatic inference of which requests can be or can't be done in parallel is tricky, and likely impossible while maintaining a semver compatible public API. Although it may not be automatic, a developer has sufficient context to tune the parallel/sequential steps. 

This RFC proposes an approach allowing a pro-user to tune the sequential vs parallel....